core/io/image_loader.h

```diff
@@ -31,47 +31,59 @@
 #ifndef IMAGE_LOADER_H
 #define IMAGE_LOADER_H
 
+#include "core/core_bind.h"
 #include "core/io/file_access.h"
 #include "core/io/image.h"
 #include "core/io/resource_loader.h"
+#include "core/object/gdvirtual.gen.inc"
 #include "core/string/ustring.h"
 #include "core/templates/list.h"
 
 class ImageLoader;
 
-class ImageFormatLoader {
+class ImageFormatLoader : public RefCounted {
+	GDCLASS(ImageFormatLoader, RefCounted);
+
 	friend class ImageLoader;
 	friend class ResourceFormatLoaderImage;
 
 protected:
-	virtual Error load_image(Ref<Image> p_image, Ref<FileAccess> p_fileaccess, uint32_t p_flags = (uint32_t)FLAG_NONE, float p_scale = 1.0) = 0;
-	virtual void get_recognized_extensions(List<String> *p_extensions) const = 0;
+	static void _bind_methods();
+
+	virtual Error load_image(Ref<Image> p_image, Ref<FileAccess> p_fileaccess, uint32_t p_flags = (uint32_t)FLAG_NONE, float p_scale = 1.0);
+	virtual void get_recognized_extensions(List<String> *p_extensions) const;
 	bool recognize(const String &p_extension) const;
 
+	GDVIRTUAL0RC(PackedStringArray, _get_recognized_extensions);
+	GDVIRTUAL4R(Error, _load_image, Ref<Image>, Ref<FileAccess>, uint32_t, float);
+
 public:
 	enum LoaderFlags {
```

Comment:
Should expose them now, I guess?

Reply:
It doesn't seem possible to expose a `BitField` as `GDVIRTUAL` if the the flags are defined in the same class (since the `VARIANT_BITFIELD_CAST` must go after the class definition, I guess this is an argument for creating a `ImageFormatLoaderExtension`)

Reply:
done